src/diff-snapshot.js

```diff
@@ -45,13 +100,16 @@ function diffImageToSnapshot(options) {
 
     const diffConfig = Object.assign({}, defaultDiffConfig, customDiffConfig);
 
-    const receivedImage = PNG.sync.read(receivedImageBuffer);
-    const baselineImage = PNG.sync.read(fs.readFileSync(baselineSnapshotPath));
-
-    if (
-      receivedImage.height !== baselineImage.height || receivedImage.width !== baselineImage.width
-    ) {
-      throw new Error('toMatchImageSnapshot(): Received image size must match baseline snapshot size in order to make comparison.');
+    let receivedImage = PNG.sync.read(receivedImageBuffer);
```

Comment:
I'd rather keep these as `const` and maybe just rename them `rawReceivedImage` or something like that like you had before. Then you can have different variables for the "processed" images that will actually be diffed by `pixelmatch`. 

I think it will make it easier to follow and debug this code in the future.